tests/test_app.py

```diff
@@ -11,7 +11,7 @@ def setUp(self):
 
     def test_decrypt_sd_user_profile(self):
         contents = self._get_file_contents("/etc/profile.d/sd-app-qubes-gpg-domain.sh")
-        expected_content = 'export QUBES_GPG_DOMAIN="sd-gpg"\n'
+        expected_content = 'if [ "$(qubesdb-read /name)" = "sd-app" ]; then export QUBES_GPG_DOMAIN="sd-gpg"; fi\n'  # noqa: E501
```

Comment:
This'll work, but to make this more of a test of behavior rather than implementation I wonder if it might be better to check if the output of `qvm-run -p sd-whonix "echo \$QUBES_GPG_DOMAIN"` is the expected `sd-gpg` -- that way we could also run the inverse test in the other AppVMs, and would be more independent of the implementation.

Reply:
good call, updated the test in f65b62c

Reply:
Nice! In case you didn't already consider and reject that in this case, note that have a `base.py` we can use to avoid duplication of test code -- see `logging_configured` as an example. Because these helper functions don't start with `test`, they won't be run as part of the test suite unless explicitly invoked with `test_*` function in child classes, as we do for the logging test.